mod/deckorate_graphql/lib/graph_q_l/types/query.rb

```diff
@@ -23,79 +34,78 @@ def singular_field fieldname, type
           end
         end
 
-        def plural_field fieldname, type
+        def plural_field fieldname, codename, type
           field fieldname, [type], null: false do
-            argument :id, Integer, required: false
             argument :limit, Integer, required: false
             argument :offset, Integer, required: false
+
+            codename.card.format.filter_keys.each do |filter|
+              if filter == :country
+                argument filter, CountryFilterType, required: false
+              elsif filter == :value_type
+                argument filter, ValueFilterType, required: false
+              elsif filter == :research_policy
+                argument filter, ResearchPolicyFilterType, required: false
+              elsif filter == :metric_type
+                argument filter, MetricCategoryFilterType, required: false
+              elsif filter == :company_category
+                argument filter, CompanyCategoryFilterType, required: false
+              elsif filter == :company_group
+                argument filter, CompanyGroupFilterType, required: false
+              elsif filter == :designer
+                argument filter, MetricDesignerFilterType, required: false
+              elsif filter == :wikirate_topic
+                argument filter, TopicFilterType, required: false
+              else
```

Comment:
I think would generalize this by making sure all the filter types map directly to the filter keys and then doing something like

```
filter_type = const_get_if_defined const("#{filter.to_s.camelize}FilterType") || String
argument filter, filter_type, required: false
```

I think you could get rid of almost all of this that way.